core/src/main/java/org/axonframework/common/lock/PessimisticLockFactory.java

```diff
@@ -131,11 +207,19 @@ public boolean isHeld() {
         }
 
         public boolean lock() {
+            if (backoffParameters.hasAcquireQueueLimit() && lock.getQueueLength() >= backoffParameters.maximumQueued) {
+                throw new LockAcquisitionFailedException("Failed to acquire lock for aggregate identifier " + identifier + ": too many queued threads.");
+            }
             try {
                 if (!lock.tryLock(0, TimeUnit.NANOSECONDS)) {
+                    int attempts = backoffParameters.acquireAttempts - 1;
```

Comment:
I think there are two reasons:

- Depending on spinTime you might hit Integer.MAX_VALUE spins quicker then you think. Which could lead to subtle bugs if you application relies on it blocking indefinitely (because you have no mechanism to recreate the command) (If you spin at 1 miliseconds you hit it at about 25 days)

- The JIT should inline hasAcquireAttemptLimit() call to false at run-time (given value of -1). Then it would eliminate the branch with the read of attempts. Which should then allow it to eliminate the attempts assignment and variable altogether.

Reply:
Great deduction @mjmbischoff! You've proven my comment invalid. Let's keep the `-1` as the value describing unlimited.